packages/workflow-core/src/lib/workflow-runner.ts

```diff
@@ -317,20 +375,109 @@ export class WorkflowRunner {
     }
 
     // Intentionally positioned after service.start() and service.send()
-    const postPlugins = this.#__extensions.statePlugins.filter(
-      plugin =>
-        plugin.isBlocking &&
-        plugin.when === 'post' &&
-        plugin.stateNames.includes(this.#__currentState),
-    );
+    const postPlugins =
+      this.#__extensions.statePlugins?.filter(
+        plugin =>
+          plugin.isBlocking &&
+          plugin.when === 'post' &&
+          plugin.stateNames.includes(this.#__currentState),
+      ) ?? [];
+    // Only iterate over the child workflows that are configured to run in the current state.
+    const childWorkflowsToInvoke =
+      this.#__childWorkflows?.filter(({ stateNames }) =>
+        stateNames?.includes(this.#__currentState),
+      ) ?? [];
 
     for (const postPlugin of postPlugins) {
       await this.#__handleAction({
         type: 'STATE_ACTION_STATUS',
         plugin: postPlugin,
-        workflowId: snapshot.machine?.id,
+        workflowId: postSendSnapshot.machine?.id,
       })(this.#__context, event);
     }
+
+    if (this.#__onInvokeChildWorkflow && childWorkflowsToInvoke?.length) {
+      const results = await Promise.allSettled(
+        childWorkflowsToInvoke?.map(
+          async ({ definitionId, runtimeId, name, version, initOptions, contextToCopy }) => {
+            try {
+              const result = await this.#__onInvokeChildWorkflow?.({
+                definitionId,
+                runtimeId,
+                name,
+                version,
+                initOptions,
+              });
+              const transformer = this.fetchTransformer(contextToCopy?.transform);
+              const data = await transformer?.transform(result as AnyRecord);
+
+              return {
+                data,
+                error: undefined,
+                runtimeId,
+              };
+            } catch (error) {
+              return {
+                data: undefined,
+                error,
+                runtimeId,
+              };
+            }
+          },
+        ),
+      );
+      const childWorkflows = results?.map(result => {
+        if (result?.status !== 'fulfilled') return;
```

Comment:
I think we should leave a todo and corresponding task to handle non fulfilled promises and errors (log them and send event to the host)

Reply:
> 

Here `try` and `catch` are in use so there shouldn't be a real `rejected` status from `allSettled`.